WMF Framework/NavigationBar.swift

```diff
@@ -18,11 +18,15 @@ public class NavigationBar: SetupView, FakeProgressReceiving, FakeProgressDelega
     public var underBarViewPercentHiddenForShowingTitle: CGFloat?
     public var title: String?
 
-    public init() {
-        super.init(frame: CGRect(x: 0, y: 0, width: 320, height: 44))
+    convenience public init() {
+        self.init(frame: CGRect(x: 0, y: 0, width: 320, height: 44))
         titleBar.frame = bounds
     }
 
+    private override init(frame: CGRect) {
```

Comment:
I don't recall overriding an `init` like this just to make it private... are there any potential issues with this?

Reply:
It goes against the idea of a designated initializer and the documentation for UIView. `It is recommended that you implement this method. You can also implement custom initialization methods in addition to, or instead of, this method.` - in this setup, that means our `init` is a convenience init and `init(frame:` is still the designated initializer and should be public. All necessary setup should be done in `init(frame:`

Reply:
🤦‍♂️

Reply:
(☕️ is helping 😂)

Reply:
f.f.r. i had some fog on swift convenience initializer specifics - these helped:
https://stackoverflow.com/a/36612507
https://stackoverflow.com/a/32108404